### PR TITLE
Data output with nokogiri

### PR DESCRIPTION
**Pull request by spdustin**
I recently refactored the `Data Output Agent` to use [NokoGiri](https://github.com/sparklemotion/nokogiri) for building the XML representation of the events. This improved speed somewhat on larger event histories, and will let us offload worrying about encoding issues to the library that everyone else uses for XML anyway.

### CODE REVIEW

**Comment by spdustin** [comment]
Updated agent picker to include agent description #830 is included in this PR, so if you end up not merging that PR (why not! :smile:) let me know and I'll cherry pick them off to another branch. My fault for basing this branch on that PR.

**Comment by cantino** [comment]
Would you mind re-merging master into this PR?

**Comment by spdustin** [comment]
Should be all set, @cantino

File: app/models/agents/data_output_agent.rb [hunk 1]
```diff
@@ -169,70 +168,57 @@ def receive_web_request(params, method, format)
...
-          content += simplify_item_for_xml(items).to_xml(skip_types: true, root: "items", skip_instruct: true, indent: 1).gsub(/^<\/?items>/, '').strip
-
-          content += Utils.unindent(<<-XML)
-            </channel>
-            </rss>
-          XML
-
-          return [content, 200, 'text/xml']
+          builder = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
+            xml.rss(:version => '2.0', 'xmlns:atom' => 'http://www.w3.org/2005/Atom') do
```
Comment (cantino) [inline app/models/agents/data_output_agent.rb]: Should we add `xmlns:media="http://search.yahoo.com/mrss/"` here too and fix https://github.com/cantino/huginn/issues/960 ?

File: app/models/agents/data_output_agent.rb [hunk 2]
```diff
@@ -169,70 +168,57 @@ def receive_web_request(params, method, format)
...
-        if @contents.is_a?(Hash)
-          options[:builder].tag! @tag_name, @attributes do
-            @contents.each { |key, value| ActiveSupport::XmlMini.to_tag(key, value, options.merge(skip_instruct: true)) }
+      data.each do |k, v|
+        if v.is_a?(Hash)
+          attrs = v.fetch('_attributes', {})
+          contents = v.fetch('_contents', v)
+          v.delete('_attributes')
+          v.delete('_contents')
+          parent.send(k, attrs) do
```
Comment (cantino) [inline app/models/agents/data_output_agent.rb]: This might not be safe.  I think `k` comes from a user (Event) and is being sent to an object.

**Pull request closed without merging**